camus-kafka-coders/src/test/java/com/linkedin/camus/etl/kafka/coders/AvroMessageDecoderTest.java

```diff
@@ -0,0 +1,122 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class AvroMessageDecoderTest {
+  private final SchemaRegistryClient schemaRegistry = new LocalSchemaRegistryClient();
+  private final KafkaAvroSerializer avroSerializer = new KafkaAvroSerializer(schemaRegistry);
+  private final KafkaAvroEncoder avroEncoder = new KafkaAvroEncoder(schemaRegistry);
+
+  private IndexedRecord createAvroRecordVersion1() {
+    String userSchema = "{\"namespace\": \"example.avro\", \"type\": \"record\", " +
+                        "\"name\": \"User\"," +
+                        "\"fields\": [{\"name\": \"name\", \"type\": \"string\"}]}";
+    Schema.Parser parser = new Schema.Parser();
+    Schema schema = parser.parse(userSchema);
+    GenericRecord avroRecord = new GenericData.Record(schema);
+    avroRecord.put("name", "testUser");
+    return avroRecord;
+  }
+
+  private IndexedRecord createAvroRecordVersion2() {
+    String userSchemaWithId = "{\"namespace\": \"example.avro\", \"type\": \"record\", " +
+                      "\"name\": \"User\"," +
+                      "\"fields\": [{\"name\": \"name\", \"type\": \"string\"}, "
+                      + "{\"name\": \"id\", \"type\": \"int\"}]}";
+    Schema.Parser parser = new Schema.Parser();
+    Schema schema = parser.parse(userSchemaWithId);
+    GenericRecord avroRecord = new GenericData.Record(schema);
+    avroRecord.put("name", "testUser");
+    avroRecord.put("id", 1);
+    return avroRecord;
+  }
+
+  private AvroMessageDecoder createAvroDecoder(
+      String topic, boolean newProducer, SchemaRegistryClient schemaRegisry ) {
+    Properties props = new Properties();
+    // Required by AvroMessageDecoder, but not needed by LocalSchemaRegistry
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+              org.apache.kafka.common.serialization.ByteArraySerializer.class);
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+              org.apache.kafka.common.serialization.ByteArraySerializer.class);
+    props.put("schema.registry.url", "http://localhost:8081");
+    if (!newProducer) {
+      props.put("is.new.producer", "false");
+    }
+    AvroMessageDecoder decoder = new AvroMessageDecoder(schemaRegisry);
+    decoder.init(props, topic);
+    return decoder;
+  }
+
+  @Test
+  public void testAvroDecoder() {
+    String topic = "testAvro";
+    Object avroRecord = createAvroRecordVersion1();
+    byte[] payload = avroSerializer.serialize(topic, avroRecord);
+    AvroMessageDecoder decoder = createAvroDecoder(topic, true, schemaRegistry);
+
+    CamusWrapper<Record> wrapper = decoder.decode(payload);
+    Object record = wrapper.getRecord();
+    assertEquals(avroRecord, record);
+
+    payload= avroEncoder.toBytes(avroRecord);
+    AvroMessageDecoder decoder2 = createAvroDecoder(topic, false, schemaRegistry);
+    wrapper = decoder2.decode(payload);
+    record = wrapper.getRecord();
+    assertEquals(avroRecord, record);
+  }
```

Comment:
It would be useful to add a test that the decoder can read a message on an older schema and project it to the latest schema (assuming it's backward compatible).